source/gfoidl.Stochastics/Statistics/SampleBuilder.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using gfoidl.Stochastics.Builders;
+
+namespace gfoidl.Stochastics.Statistics
+{
+    public class SampleBuilder
+    {
+        // Must not be readonly, cf. https://gist.github.com/gfoidl/14b07dfe8ee5cb093f216f8a85759d88
+        private ArrayBuilder<double> _arrayBuilder;
+        private double _min = double.MaxValue;
+        private double _max = double.MinValue;
+        private double _sum;
+        //---------------------------------------------------------------------
+        public SampleBuilder() => _arrayBuilder = new ArrayBuilder<double>(true);
+        //---------------------------------------------------------------------
+        public void Add(double item)
+        {
+            _arrayBuilder.Add(item);
+
+            _sum += item;
+            if (item < _min) _min = item;
+            if (item > _max) _max = item;
+        }
+        //---------------------------------------------------------------------
+        public void Add(IEnumerable<double> values)
+        {
+            double min = double.MaxValue;
+            double max = double.MinValue;
+            double sum = 0;
+
+            foreach (double item in values)
+                this.AddCore(item, ref min, ref max, ref sum);
+
+            _min = min;
+            _max = max;
+            _sum = sum;
+        }
+        //---------------------------------------------------------------------
+        public IEnumerable<double> AddWithYield(IEnumerable<double> values)
```

Comment:
Maybe also the shortcut (see above)? This method could simple return `Sample` in that case, because sample is a _IEnumerable<double>_.